### PR TITLE
Ashwalker Update

### PR DESCRIPTION
**Pull request by SakuraOran**
## About The Pull Request
Ashwalker code is very old, and this aims to bring it up to speed. It adds tendril revival mechanics, allowing ashwalkers to heal by sleeping next to the tendril, and even reviving dead ashwalkers by bringing their corpse to the tendril. It also fixes an issue where some items that are supposed to drop on butcher don't get dropped when butchered by the tendril.

## Why It's Good For The Game
Ghost roles that actually work properly is never a bad thing

## Changelog
:cl:
add: Ashwalkers can now heal, even from death, by being incapacitated (asleep/dead) near the tendril
add: Added the ability for antags to have the "protect object" objective, which the ashwalkers now have to protect the tendril. This is determined by if the object is broken or not, and can be set to any object or structure that can reasonably be destroyed
balance: Ashwalkers get one life, and have a 40% chance to earn an additional life when they sacrifice to the tendril.
fix: fixed mobs not dropping guaranteed_butcher_results when gibbed 
/:cl:

### CODE REVIEW

File: code/datums/mood_events/generic_positive_events.dm [hunk 1]
```diff
@@ -205,3 +205,8 @@
 /datum/mood_event/cleared_stomach
 	description = "<span class='nicegreen'>Feels nice to get that out of the way!</span>\n"
 	mood_change = 3
+
+/datum/mood_event/sacrifice_good
+	description = "<span class='nicegreen'>The gods are pleased with this offering!</span>\n"
+	mood_change = 5
+	timeout = 3 MINUTES
```
Comment (Putnam3145) [inline code/datums/mood_events/generic_positive_events.dm]: add a newline here

File: code/modules/antagonists/ashwalker/ashwalker.dm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/datum/antagonist/ashwalker/on_removal()
+	. = ..()
+	UnregisterSignal(owner.current, COMSIG_MOB_EXAMINATE)
+
+/datum/antagonist/ashwalker/proc/on_examinate(datum/source, atom/A)
+	SIGNAL_HANDLER
+
+	if(istype(A, /obj/structure/headpike))
+		SEND_SIGNAL(owner.current, COMSIG_ADD_MOOD_EVENT, "oogabooga", /datum/mood_event/sacrifice_good)
```
Comment (Putnam3145) [inline code/modules/antagonists/ashwalker/ashwalker.dm]: add a newline here

**Review by Putnam3145 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 3]
```diff
@@ -36,6 +36,44 @@
...
+
+/obj/structure/ash_walker_eggshell/attack_ghost(mob/user) //Pass on ghost clicks to the mob spawner
+	if(egg)
+		egg.attack_ghost(user)
+	. = ..()
+
+/obj/structure/ash_walker_eggshell/Destroy()
+	if(!egg)
+		return ..()
+	var/mob/living/carbon/human/yolk = new /mob/living/carbon/human/(get_turf(src))
```
Comment (Putnam3145) [inline code/game/objects/structures/ghost_role_spawners.dm:67]: does this work??? that / at the end is odd

File: code/modules/antagonists/_common/antag_datum.dm [hunk 4]
```diff
@@ -59,7 +59,7 @@ GLOBAL_LIST_EMPTY(antagonists)
 /datum/antagonist/proc/on_body_transfer(mob/living/old_body, mob/living/new_body)
 	SHOULD_CALL_PARENT(TRUE)
 	remove_innate_effects(old_body)
-	if(old_body.stat != DEAD && !LAZYLEN(old_body.mind?.antag_datums))
+	if(old_body && old_body.stat != DEAD && !LAZYLEN(old_body.mind?.antag_datums))
```
Comment (Putnam3145) [inline code/modules/antagonists/_common/antag_datum.dm]:
```suggestion
	if(old_body && old_body.stat != DEAD && !length(old_body.mind?.antag_datums))
```

File: code/modules/antagonists/ashwalker/ashwalker.dm (hunk 2 above)
Comment (Putnam3145) [inline code/modules/antagonists/ashwalker/ashwalker.dm]: i feel like i should probably 1984 that mood cause string

File: code/game/objects/structures/ghost_role_spawners.dm (hunk 3 above)
Comment (SakuraOran) [reply code/game/objects/structures/ghost_role_spawners.dm:67]: uhhhhhhh i may have forgot to test that part but i think so. ill get back you on that

Comment (SakuraOran) [reply code/game/objects/structures/ghost_role_spawners.dm:67]: https://gyazo.com/fcad061c28a8f12a078a1364a2552a4f confirmed, does work

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 5]
```diff
@@ -55,12 +93,25 @@
...
 	Ensure your nest remains protected at all costs."
 	assignedrole = "Ash Walker"
+	var/datum/team/ashwalkers/team
+	var/obj/structure/ash_walker_eggshell/eggshell
+
+/obj/effect/mob_spawn/human/ash_walker/Destroy()
+	eggshell = null
+	return ..()
+
+/obj/effect/mob_spawn/human/ash_walker/allow_spawn(mob/user, silent = FALSE)
```
Comment (DeltaFire15) [inline code/game/objects/structures/ghost_role_spawners.dm:103]: I disagree with limiting ashwalker spawns to one per player as ashwalker is often a dangerous and lone role, with there not really being any reason to deny it. (Even if acquiring a respawn is not too difficult if RNG is on your side when sacrificing)

Comment (SakuraOran) [reply code/game/objects/structures/ghost_role_spawners.dm:103]: also remember that you start with 2 bodies in the nest. you can also be revived by being dragged to the tentacle. its meant to be a team thing, and the focus is being, get to the tentacle and you come back to full health. i think the having it limited in respawns balances that

Comment (Putnam3145) [reply code/game/objects/structures/ghost_role_spawners.dm:103]: we kinda have to take into consideration when there's not enough players for the team thing to work out; there's a lot of solo ashwalker play here

**Review by Putnam3145 — APPROVED** [review]

**Pull request merged**